src/compiler/types.rs

```diff
@@ -2756,7 +2884,28 @@ impl DiagnosticRelatedInformationInterface for BaseDiagnostic {
     }
 }
 
+impl DiagnosticInterface for BaseDiagnostic {
+    fn maybe_related_information(&self) -> Option<&[DiagnosticRelatedInformation]> {
+        self.related_information.as_deref()
```

Comment:
What does `as_deref` do (vs I guess `*`)?

Reply:
Here are the [docs](https://doc.rust-lang.org/std/option/enum.Option.html#method.as_deref), I guess it's like `.as_ref()` (on an `Option`) but also deref's the now-referenced content